netty/src/test/java/io/grpc/netty/NettyTransportTest.java

```diff
@@ -106,4 +114,48 @@ protected ManagedClientTransport newClientTransport(InternalServer server) {
   public void clientChecksInboundMetadataSize_trailer() throws Exception {
     // Server-side is flaky due to https://github.com/netty/netty/pull/8332
   }
+
+  @Test
+  public void channelHasUnresolvedHostname() throws Exception {
+    server = null;
+    final SettableFuture<Status> future = SettableFuture.create();
+    ChannelLogger logger = transportLogger();
+    ManagedClientTransport transport = clientFactory.newClientTransport(
+        InetSocketAddress.createUnresolved("invalid", 1234),
+        new ClientTransportFactory.ClientTransportOptions()
+            .setChannelLogger(logger), logger);
+    Runnable runnable = transport.start(new ManagedClientTransport.Listener() {
+      final Throwable failTestException =
+          new Throwable("transport should have failed and shutdown but didnt");
+      @Override
+      public void transportShutdown(Status s) {
+        future.set(s);
+      }
+
+      @Override
+      public void transportTerminated() {}
+
+      @Override
+      public void transportReady() {
+        future.setException(failTestException);
+      }
+
+      @Override
+      public void transportInUse(boolean inUse) {
+        future.setException(failTestException);
+      }
+    });
+    if (runnable != null) {
+      runnable.run();
+    }
+    try {
+      Status status = future.get();
+      assertEquals(Status.Code.UNAVAILABLE, status.getCode());
+      assertTrue(status.getCause() instanceof UnresolvedAddressException);
```

Comment:
You can leave it as-is, but I'll note that assertTrue/assertFalse commonly produce poor error messages if they fail.

This is one of the biggest reasons using Truth can be nice. With Truth, this would be `assertThat(status.getCause()).isinstanceof(UnresolvedAddressException.class)` (`assertThat` is typically a static import from the `Truth` class). Then if it fails, instead of saying "expected true but was false" it can say "expected UnresolvedAddressException but was SomeOtherException" which can greatly aid debugging. It's possible to do that with the JUnit assertion, but you have to create a string and pass it as the first argument. Something to consider for the future.

Reply:
I like that a lot better! Change made